src/vm.rs

```diff
@@ -122,16 +148,16 @@ impl InstructionMeter for DefaultInstructionMeter {
 /// ];
 ///
 /// // Instantiate a VM.
-/// let executable = EbpfVm::<UserError>::create_executable_from_text_bytes(prog, None).unwrap();
-/// let mut vm = EbpfVm::<UserError>::new(executable.as_ref(), mem, &[]).unwrap();
+/// let executable = Executable::<UserError>::from_text_bytes(prog, None).unwrap();
+/// let mut vm = EbpfVm::<UserError, DefaultInstructionMeter>::new(executable.as_ref(), mem, &[]).unwrap();
 ///
 /// // Provide a reference to the packet data.
 /// let res = vm.execute_program_interpreted(&mut DefaultInstructionMeter {}).unwrap();
 /// assert_eq!(res, 0);
 /// ```
-pub struct EbpfVm<'a, E: UserDefinedError> {
+pub struct EbpfVm<'a, E: UserDefinedError, I: InstructionMeter> {
     executable: &'a dyn Executable<E>,
-    compiled_prog: Option<JitProgram<E>>,
+    compiled_prog_and_arg: Option<(JitProgram<E, I>, Vec<*const u8>)>,
```

Comment:
I might have already asked this but why `and_arg` and not just `compiled_prog`?

Reply:
It is a merge from https://github.com/solana-labs/rbpf/pull/100
The reason is that the argument is a concat of the memory mapping (which is specific to a VM instance) and the instruction offsets (which are VM independent). But both can be kept over multiple invocations (saves memcpy). Not sure if that is an actual use-case, can revert it if it causes too much trouble.

Reply:
Let's leave it, maybe we'll switch the bank to caching VM instances rather than executables :-).